relay-sampling/src/lib.rs

```diff
@@ -108,40 +108,48 @@ mod tests {
         DecayingFunction, RuleId, RuleType, SamplingMode, SamplingRule, SamplingValue, TimeRange,
     };
     use crate::dsc::TraceUserContext;
-    use crate::evaluation::{merge_configs_and_match, MatchedRuleIds, SamplingMatch};
+    use crate::evaluation::{
+        get_sampling_match, match_rules, sampling_match, MatchedRuleIds, SamplingResult,
+    };
 
     use super::*;
 
-    macro_rules! assert_transaction_match {
-        ($res:expr, $sr:expr, $sd:expr, $( $id:expr ),*) => {
-            assert_eq!(
-                $res,
-                Some(SamplingMatch {
-                    sample_rate: $sr,
-                    seed: $sd.id.value().unwrap().0,
-                    matched_rule_ids: MatchedRuleIds(vec![$(RuleId($id),)*])
-                })
-            )
+    fn sampling_match_from_parts(
+        sample_rate: f64,
+        seed: Uuid,
+        matched_rules: MatchedRuleIds,
+    ) -> SamplingResult {
+        let is_kept = sampling_match(sample_rate, seed);
+        SamplingResult::Match {
+            sample_rate,
+            seed,
+            matched_rules,
+            is_kept,
         }
     }
 
-    macro_rules! assert_trace_match {
-        ($res:expr, $sr:expr, $sd:expr, $( $id:expr ),*) => {
-            assert_eq!(
-                $res,
-                Some(SamplingMatch {
-                    sample_rate: $sr,
-                    seed: $sd.trace_id,
-                    matched_rule_ids: MatchedRuleIds(vec![$(RuleId($id),)*])
-                })
-            )
-        }
+    fn transaction_match(
```

Comment:
These are now just helpers to construct transaction and trace SamplingResults, i found it clearer to do it like this and then use the normal assert_eq in the tests